CMakeLists.txt

```diff
@@ -310,12 +310,16 @@ add_custom_target(check-format ${BUILD_SUPPORT_DIR}/run_clang_format.py
 # "make check-clang-tidy" target
 ###########################################################
 if (${CLANG_TIDY_FOUND})
-    # runs clang-tidy and exits with a non-zero exit code if any errors are found.
+    set(CLANG_TIDY_THREADS 4)
```

Comment:
Is there a reason to set it to 4 instead of 0?
I think 0 will use all cores, [line 201](https://github.com/cmu-db/terrier/blob/8db01ce24341b6f9b593faf3b9d1e6453d8b7ee0/build-support/clang-tidy-diff.py#L201).